ext/opcache/ZendAccelerator.c

```diff
@@ -3300,6 +3303,39 @@ static zend_result accel_post_startup(void)
 #endif
 		accel_shared_globals = calloc(1, sizeof(zend_accel_shared_globals));
 	}
+
+	/* opcache.file_cache_read_only should only be enabled when all script files are read-only */
+	if (ZCG(accel_directives).file_cache_read_only) {
+		if (!ZCG(accel_directives).file_cache) {
+			accel_startup_ok = false;
+			zend_accel_error_noreturn(ACCEL_LOG_FATAL, "opcache.file_cache_read_only is set without a proper setting of opcache.file_cache");
+			return SUCCESS;
+		}
+		if (ZCG(accel_directives).revalidate_freq != 0) {
+			accel_startup_ok = false;
+			zend_accel_error_noreturn(ACCEL_LOG_FATAL, "opcache.file_cache_read_only cannot be enabled when opcache.revalidate_freq is not 0.");
+			return SUCCESS;
+		}
+		if (ZCG(accel_directives).validate_timestamps) {
+			accel_startup_ok = false;
+			zend_accel_error_noreturn(ACCEL_LOG_FATAL, "opcache.file_cache_read_only cannot be enabled when opcache.validate_timestamps is enabled.");
+			return SUCCESS;
+		}
```

Comment:
I think, these checks may be removed.
In case these options are set to 0, opcache will always prefer file_cache, otherwise it'll compile PHP script if it's newer than cached.